migrate/20231207_storage_device.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    create_table(:storage_device) do
+      column :id, :uuid, primary_key: true
+      column :name, :text, null: false
+      column :total_storage_gib, :Integer, null: false
+      column :available_storage_gib, :Integer, null: false
+      column :enabled, :bool, null: false, default: true
+      foreign_key :vm_host_id, :vm_host, type: :uuid
```

Comment:
I am wondering if we should add a jsonb flags field, to fill it with things like `{filesystem: 'btrfs'}`. Some of the code (e.g. forcing btrfs device if not using bdev_ubi) might be cleaner if we had this. But it also depends on when we are going to drop the btrfs support completely.

Not sure what other usecases we will have in future.

(This is mostly a thought, which can be ignored now & revisited in future after we have seen more usecases)

Reply:
Adding a column isn't so hard at a later time … but it will affect the allocation algorithm and programming interface, which is the crux of this patch …